crates/sui-graphql-rpc/src/types/type_filter.rs

```diff
@@ -330,21 +394,211 @@ mod tests {
         expect.assert_eq(&format!("{filters:#?}"))
     }
 
+    #[test]
+    fn test_valid_function_filters() {
+        let inputs = [
+            "0x2",
+            "0x2::coin",
+            "0x2::object::new",
+            "0x2::tx_context::TxContext",
+        ]
+        .into_iter();
+
+        let filters: Vec<_> = inputs.map(|i| FqNameFilter::from_str(i).unwrap()).collect();
+
+        let expect = expect![[r#"
+            [
+                ByModule(
+                    ByPackage(
+                        SuiAddress(
+                            [
+                                0,
+                                0,
+                                0,
+                                0,
+                                0,
+                                0,
+                                0,
+                                0,
+                                0,
+                                0,
+                                0,
+                                0,
+                                0,
+                                0,
+                                0,
+                                0,
+                                0,
+                                0,
+                                0,
+                                0,
+                                0,
+                                0,
+                                0,
+                                0,
+                                0,
+                                0,
+                                0,
+                                0,
+                                0,
+                                0,
+                                0,
+                                2,
+                            ],
+                        ),
+                    ),
+                ),
+                ByModule(
+                    ByModule(
+                        SuiAddress(
+                            [
+                                0,
```

Comment:
I thought you had introduced a PR some time ago that rendered this in a different format, in 0x, instead of an array. Is it possible to apply it here?

Reply:
Let me see if I can serialize it as JSON (I'm not sure this will work, but it's worth a try!), or otherwise I'll try the non-indented debug format.  I have some further changes to this file coming up, so I will handle this in a follow-up PR.

Note that this is just the debug output, for testing purposes, so this format is not something that clients will be exposed to.

Reply:
I ended up implementing `Display` for these types for these tests instead, thanks for the suggestion!

Reply:
#15713